src/internal/packager2/layout/create.go

```diff
@@ -260,7 +260,8 @@ func CreateSkeleton(ctx context.Context, packagePath string, opt CreateOptions)
 	return buildPath, nil
 }
 
-func loadPackage(ctx context.Context, packagePath, flavor string, setVariables map[string]string) (v1alpha1.ZarfPackage, error) {
+// LoadPackage returns a validated package definition after flavors, imports, and variables are applied.
+func LoadPackage(ctx context.Context, packagePath, flavor string, setVariables map[string]string) (v1alpha1.ZarfPackage, error) {
```

Comment:
You should use the following function instead.

https://github.com/zarf-dev/zarf/blob/15216b5480661e1284ab1338d9897fe2403a0c6f/src/internal/packager2/load.go#L38

The API may need to be cleaned up in the future.